src/lib/operators/join_index.cpp

```diff
@@ -208,7 +238,13 @@ void JoinIndex::_join_two_segments_using_index(LeftIterator left_it, LeftIterato
     auto range_begin = BaseIndex::Iterator{};
     auto range_end = BaseIndex::Iterator{};
 
-    switch (_primary_predicate.predicate_condition) {
+    auto primary_predicate = _primary_predicate;
+
+    if (_index_side == JoinInputSide::Left) {
+      primary_predicate.flip();
```

Comment:
The predicate should be flipped once and then stored. We shouldn't have flipped and non-flipped versions flying around.